oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxSoundCard.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Oshi (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2018 The Oshi Project Team
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ *
+ * Contributors:
+ * https://github.com/oshi/oshi/graphs/contributors
+ */
+package oshi.hardware.platform.linux;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import oshi.hardware.common.AbstractSoundCard;
+import oshi.util.FileUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Sound card data obtained via /proc/asound directory
+ *
+ * @author BilalAM
+ */
+
+public class LinuxSoundCard extends AbstractSoundCard {
+
+        private static final String SC_PATH = "/proc/asound/";
+        private static final String CARD_FOLDER = "card";
+        private static final String CARDS_FILE = "cards";
+        private static final String ID_FILE = "id";
+
+        private static final Logger LOG = LoggerFactory.getLogger(LinuxSoundCard.class);
+
+        public LinuxSoundCard(String kernelVersion, String name, String codec) {
+                super(kernelVersion, name, codec);
+        }
+
+        /**
+         * Method to find all the card folders contained in the <b>asound</b> folder denoting the cards
+         * currently contained in our machine.
+         *
+         * @return : A list of files starting with 'card'
+         */
+        private static List<File> getCardFolders() {
+                File cardsDirectory = new File(SC_PATH);
+                List<File> cardFolders = new ArrayList<>();
+                for (File card : Objects.requireNonNull(cardsDirectory.listFiles())) {
```

Comment:
I think this is where your NPE is being generated.  `requireNonNull()` throws NPE if the argument is null.  You would have to catch this exception and skip the iteration (or better yet have a temporary variable for the file list and null check it!).

Reply:
And also I will fix the rest of the things first thing tomorrow morning at the earliest (some unused imports and debugging code and also the format)